### PR TITLE
Configure both channel IDs for aie_trace & aie_profile

### PR DESCRIPTION
**Pull request by vipangul**
#### Problem solved by the commit
This change is for [CR-1194437]
CR mentions that channel IDs were not selected appropriately although specified in xrt.ini configuration

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
code flow doesn't set the channelIDs and related config correctly.

#### How problem was solved, alternative solutions (if any) and why they were rejected
Modified code to handle following:
-  Identify which channel IDs are specified and allot them to available channelId0 & channelId1
- It is possible that only single channelID is specified, we still mark channelId0 & channelId1 which further gets considered as duplicate.

#### Risks (if any) associated the changes in the commit
Different combination of metric switches needs to be verified.

#### What has been tested and how, request additional testing if necessary

Tested on vck190 for basic testcase of 
```
 tile_based_interface_tile_metrics=all:output_ports_stalls:0
 tile_based_interface_tile_metrics=all:output_ports_stalls:1
```
 Visualized the trace data on vitis analyzer.
 Applicable changes for aie_profile plugin too.
 
 **Pending testcases/issues & expected behavior:**
 ```
  tile_based_interface_tile_metrics=<minCol>:<maxCol>:output_ports_stalls:0
 tile_based_interface_tile_metrics=<singleCol>:output_ports_stalls:1
 ```

### CODE REVIEW

File: src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp [hunk 1]
```diff
@@ -772,25 +772,33 @@ namespace xdp {
...
+    // all:<metric>[:<channel0>[:<channel1>]]
     for (size_t i = 0; i < metricsSettings.size(); ++i) {
       // Split done only in Pass 1
       boost::split(metrics[i], metricsSettings[i], boost::is_any_of(":"));
 
       // Check if not all tiles or invalid format
       if ((metrics[i][0].compare("all") != 0) || (metrics[i].size() < 2))
         continue;
+      
+      std::cout<<"!!! P1 MetricsSettings: "<<metricsSettings[i]<<std::endl;
```
Comment (pgschuey) [inline src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp:784]: please remove this debug message

File: src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp [hunk 2]
```diff
@@ -804,6 +812,7 @@ namespace xdp {
         continue;
       }
 
+      std::cout<<"!!! P2 MetricsSettings: "<<metricsSettings[i]<<std::endl;
```
Comment (pgschuey) [inline src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp:815]: please remove this message too

File: src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp [hunk 3]
```diff
@@ -833,20 +844,23 @@ namespace xdp {
...
     }  // Pass 2
 
     // Pass 3 : process only single tile metric setting
+    // <singleColumn>:<metric>[:<channel0>[:<channel1>]]
     for (size_t i = 0; i < metricsSettings.size(); ++i) {
       // Skip if already processed or invalid format
       if ((processed.find(i) != processed.end()) || (metrics[i].size() < 2))
         continue;
 
+      std::cout<<"!!! P3 MetricsSettings: "<<metricsSettings[i]<<std::endl;
```
Comment (pgschuey) [inline src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp:863]: remove this please

**Review by pgschuey — COMMENTED** [review]
please also cover channel parsing in graph-based settings, too

**Review by pgschuey — APPROVED** [review]

**Review by jvillarre — APPROVED** [review]

**Pull request merged**